* Emacs should now include ZWJs and ZWNJs for all Indic scripts.
  Patch: use-zwj-and-zwnj-for-indic-scripts.diff
  Originally-reported-by: Praveen A <pravi.a@gmail.com>
  Date: Thu, 3 Dec 2009 11:51:02 UTC
  Added-by: Rob Browning <rlb@defaultvalue.org>
  Status: incorporated upstream
  Bug: 559292

  The Debian patch is taken from this upstream commit:

    From f0c7c513b71778e3767be145334e20872f6c68be Mon Sep 17 00:00:00 2001
    From: Kenichi Handa <handa@m17n.org>
    Date: Wed, 2 Dec 2009 07:59:51 +0000
    Subject: [PATCH] Include ZWJ and ZWNJ in the patterns to shape for all Indic scripts.

Index: b/lisp/language/indian.el
===================================================================
--- a/lisp/language/indian.el
+++ b/lisp/language/indian.el
@@ -165,15 +165,15 @@
   "Regexp matching a composable sequence of Malayalam characters.")
 
 (let ((script-regexp-alist
-       `((devanagari . ,devanagari-composable-pattern)
+       `((devanagari . "[\x900-\x97F\x200C\x200D]+")
 	 (bengali . "[\x980-\x9FF\x200C\x200D]+")
 	 (gurmukhi . "[\xA00-\xA7F\x200C\x200D]+")
 	 (gujarati . "[\xA80-\xAFF\x200C\x200D]+")
 	 (oriya . "[\xB00-\xB7F\x200C\x200D]+")
-	 (tamil . ,tamil-composable-pattern)
+	 (tamil . "[\xB80-\xBFF\x200C\x200D]+")
 	 (telugu . "[\xC00-\xC7F\x200C\x200D]+")
-	 (kannada . ,kannada-composable-pattern)
-	 (malayalam . ,malayalam-composable-pattern))))
+	 (kannada . "[\xC80-\xCFF\x200C\x200D]+")
+	 (malayalam . "[\xD00-\xD7F\x200C\x200D]+"))))
   (map-char-table
    #'(lambda (key val)
        (let ((slot (assq val script-regexp-alist)))
